src/htm/algorithms/Connections.hpp

```diff
@@ -361,6 +361,33 @@ class Connections : public Serializable
                         const CellIdx presynapticCell,
                         Permanence permanence);
 
+
+
+  /**
+   * Grow new synapses for disconnected inputs/candidates.
+   *
+   * For each specified segments, grow synapses to all specified inputs that
+   * aren't already connected to the segment. 
+   *
+   * @param segment - Segment: The segment to modify 
+   * @param growthCandidates - vector<Synapse>&: The inputs to connect to 
+   * @param initialPermanence - Permanence: The permanence for each added synapse
+   * @param maxNew - (optional) size_t, default = 0/unused. If set to [1, growthCandidates.size())
+   *   then we subsample the disconnected inputs to be connected. This is used to limit "bursts" when
+   *   a huge number of cells/inputs would be newly connected at once. Which would disturb the HTM.
+   * @param rng - Random&, used to sample if `maxNew` is used. 
+   * @param maxSynapsesPerSegment - (optional) size_t, default=0/off. If >0: enforce limit on max
+   *   number of synapses on a segment. If reached, weak synapses will be purged to make space.
+   *
+   **/
+  void growSynapses(const Segment segment, 
```

Comment:
Is this something that we should expose to NetworkAPI?
I know this is very low level internal stuff and I would rather not unless you think it would be useful.

Reply:
> Is this something that we should expose to NetworkAPI? 

good point, and I agree not to unless necessary/requested. NetworkAPI is based around regions and no region uses/exposes this functionality, so this is not needed.